### PR TITLE
Add loading, match, no-match, and error page views and persist user page view state via local storage

### PR DESCRIPTION
**Pull request by dhruvi618**
Files changed:
- App.js: Persist user workflow state to local storage, add useEffect React hook to load page view from local storage and functionality to conditionally render page views based on current user workflow state. 
- ErrorPage.js: Add ErrorPage component with content to notify user that an unexpected error has occurred
- Form.js: Change default state of match preference from none to any
- FormContent.js: Modularize form content that was initially in Form.js to a separate file for easy readability
- LoadingPage.js: Add LoadingPage component with content to notify user that the matching algo is currently in process
- MatchPage.js: Add LoadingPage component with content to notify user that a match has been found
- NoMatchPage.js: Add LoadingPage component with content to notify user that a match could not be found

Snapshot file changed:
- App.test.js.snap: Update snapshot corresponding to multiple page views

Screenshots of views:
- Form: https://screenshot.googleplex.com/DhLwJBmNgS9
- Loading: https://screenshot.googleplex.com/96HsoAJBdfV
- Match: https://screenshot.googleplex.com/wNTMX3H2WEf
- NoMatch: https://screenshot.googleplex.com/U5QKfztEqwO
- Error: https://screenshot.googleplex.com/k2dnU1pCVkH

Tests included in a separate PR!

### CODE REVIEW

File: frontend/src/App.js [hunk 1]
```diff
@@ -51,52 +55,95 @@ export async function fetchMatch() {
...
+              />
+            </Card>
+          </div>
+        </div>
+      );
+    case 'loading':
+      return (
+        <div>
+          <MenuBar />
+          <div className={classes.centerHorizontal}>
```
Comment (TcgVanguardTroll) [inline frontend/src/App.js:114]: Nice use of switch statement her 😃

File: frontend/src/__snapshots__/App.test.js.snap [hunk 2]
```diff
@@ -58,43 +58,45 @@ exports[`App should render correctly 1`] = `
...
+            <em>
+              now
+            </em>
+            !
+          </h3>
+          <p>
+            Miss bumping into new faces at the office? Want an easy, fun, spontaneous way of meeting Googlers virtually? Now you can!
+          </p>
+          <p>
+            Ad-lib matches you with a fellow Googler in the queue, notifies you through email when you’ve been matched, and adds an event to your Calendar with a Meet link for you to join immediately! It also provides a starter question to get the conversation flowing!
```
Comment (jzwang8) [inline frontend/src/__snapshots__/App.test.js.snap:79]: Are we still doing the starter question?

File: frontend/src/components/ErrorPage.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+ * Define ErrorPage component
+ * @return {ErrorPage} ErrorPage component
+ */
+export default function ErrorPage() {
+  const classes = useStyles();
+  return (
+    <div>
+      <Card className={classes.content}>
+        <CardContent>
+          <h3>Oops, something went wrong</h3>
```
Comment (jzwang8) [inline frontend/src/components/ErrorPage.js:40]: ! after

File: frontend/src/components/ErrorPage.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+          <p>We encourage you to please try again later!</p>
```
Comment (jzwang8) [inline frontend/src/components/ErrorPage.js:41]: Maybe "We apologize for the inconvenience. Please try again later!"

File: frontend/src/components/MatchPage.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+export default function MatchPage(props) {
+  const classes = useStyles();
+  return (
+    <div>
+      <Card className={classes.content}>
+        <CardContent>
+          <h3>We found you a match!</h3>
+          {/* TO-DO(#78): Display more personalized info on page*/}
+          <p>This Googler is so excited to meet you.</p>
+          <p>Please check your Calendar to find the event and Meet link and
```
Comment (jzwang8) [inline frontend/src/components/MatchPage.js:44]: period after

File: frontend/src/App.js [hunk 6]
```diff
@@ -51,52 +55,95 @@ export async function fetchMatch() {
...
+  /**
+   * Set current page view to state retrieved from local storage or
+   * default view if no state is currently saved
+   * */
+  const [currentPage, setCurrentPage] = React.useState(
+      localStorage.getItem(pageViewKey) || defaultPageView,
+  );
+
+  // Load page view state from local storage using useEffect hook
+  useEffect(() => {
```
Comment (TcgVanguardTroll) [inline frontend/src/App.js:73]: I'm curious , can you explain what this line is doing ❓

File: frontend/src/components/NoMatchPage.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+export default function NoMatchPage(props) {
+  const classes = useStyles();
+  return (
+    <div>
+      <Card className={classes.content}>
+        <CardContent>
+          <h3>We could not find you a match!</h3>
+          {/* TO-DO(#78): Display more personalized info on page*/}
+          <p>It looks like you are only free until 11:45am, and we could not
+             find you a match to meet or 30 minutes before then. Please try
```
Comment (jzwang8) [inline frontend/src/components/NoMatchPage.js:44]: to meet for. Also how would you get the endTimeAvailable and duration? Should the backend return endTimeAvailable and duration when no match response is sent?

File: frontend/src/components/NoMatchPage.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+ * @param {Object} props
+ * @return {NoMatchPage} NoMatchPage component
+ */
+export default function NoMatchPage(props) {
+  const classes = useStyles();
+  return (
+    <div>
+      <Card className={classes.content}>
+        <CardContent>
+          <h3>We could not find you a match!</h3>
```
Comment (jzwang8) [inline frontend/src/components/NoMatchPage.js:41]: "Sorry, we couldn't find you a match :(" or "Bummer, we couldn't find you a match!" aka something less formal

File: frontend/src/components/MatchPage.js [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+          join to meet your new friend</p>
+        </CardContent>
+      </Card>
+    </div>
+  );
+}
```
Comment (TcgVanguardTroll) [inline frontend/src/components/MatchPage.js:50]: Will screenshots of these new pages be added some-time later in the PR ?

Comment (TcgVanguardTroll) [reply frontend/src/components/MatchPage.js:50]: NVM found them after writing my comment 😅

File: frontend/src/components/LoadingPage.js [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+  content: {
+    margin: theme.spacing(2),
+    width: 800,
+  },
+}));
+
+/**
+ * Define LoadingPage component
+ * @return {LoadingPage} LoadingPage component
+ */
```
Comment (TcgVanguardTroll) [inline frontend/src/components/LoadingPage.js:34]: Do you think its possible to add a loading gif in the center of the page ?

File: frontend/src/App.js (hunk 1 above)
Comment (dhruvi618) [reply frontend/src/App.js:114]: Thank you!

File: frontend/src/components/NoMatchPage.js (hunk 7 above)
Comment (dhruvi618) [reply frontend/src/components/NoMatchPage.js:44]: Yeah, I think that would be good to include in the expired JSON object for a more personalized experience, what do you think?

File: frontend/src/__snapshots__/App.test.js.snap (hunk 2 above)
Comment (dhruvi618) [reply frontend/src/__snapshots__/App.test.js.snap:79]: Hmm, good point, I am not if we will have to implement this. @TcgVanguardTroll  what do you think about this? If not, I can take it out of the description.

File: frontend/src/components/MatchPage.js (hunk 9 above)
Comment (dhruvi618) [reply frontend/src/components/MatchPage.js]: I have added these in the PR description above

File: frontend/src/components/LoadingPage.js (hunk 10 above)
Comment (dhruvi618) [reply frontend/src/components/LoadingPage.js:34]: Sure! Is there a particular GIF you had in mind? I can add an issue for this and add in a follow up PR

File: frontend/src/__snapshots__/App.test.js.snap (hunk 2 above)
Comment (dhruvi618) [reply frontend/src/__snapshots__/App.test.js.snap:79]: As per our discussion during standup, deciding to keep this in for now and will add in functionality for it in a follow up PR.

File: frontend/src/App.js [hunk 11]
```diff
@@ -51,52 +55,95 @@ export async function fetchMatch() {
  * @return {App} App component
  */
 export default function App() {
+  let match;
```
[the remark this answers is not in the document]
Comment (dhruvi618) [reply frontend/src/App.js]: Good point, changed this to be a global variable that can be accessed and communicated down to the child components.

**Pull request merged**